### ISSUE TITLE
[feature] Add API for event tracing

### ISSUE THREAD

**Issue by QGIS-feature-tracker** [qgis/QGIS-Documentation#4359]
Original commit: https://github.com/qgis/QGIS/commit/a192268f52ddd822e9aa654abfaf3ef75ee232d8 by wonder-sk

This is more developer oriented feature to help with profiling, debugging
and optimization - especially when dealing with multi-threaded code.

**Comment by DelazJ** [comment]
@roya0045 I don't think this is a PyQGIS Cookbook issue. Maybe something for the [Developers guide](https://docs.qgis.org/testing/en/docs/developers_guide/index.html) but I wonder if it is still worth it. @wonder-sk, any insights please? Thanks

**Comment by roya0045** [comment]
> @roya0045 I don't think this is a PyQGIS Cookbook issue. Maybe something for the [Developers guide](https://docs.qgis.org/testing/en/docs/developers_guide/index.html) but I wonder if it is still worth it. @wonder-sk, any insights please? Thanks

Since it's in the api I assume this could be good in the cookbook, but that was just my assumption.
